apps/extension/src/provider/Signer.ts

```diff
@@ -76,6 +79,18 @@ export class Signer implements ISigner {
     return await this._anoma.encodeBonding(toBase64(serializedTransfer));
   }
 
+  /**
+   * Submit bond transaction
+   */
+  public async submitBond(args: SubmitBondProps): Promise<void> {
+    const msgValue = new SubmitBondMsgValue(args);
+
+    const msg = new Message<SubmitBondMsgValue>();
+    const encoded = msg.encode(SubmitBondMsgSchema, msgValue);
```

Comment:
We are encoding this here using Borsh schema, but I thought that this should be a responsibility of the SDK?

Reply:
... this https://github.com/anoma/namada/blob/e40e91a0ae78f33870482f48535a7ae12d94e3f2/shared/src/ledger/tx.rs#L725

Reply:
Not sure. I think it is done like that because CLI does not need to serialize structs.